documentation/migration-guides/Stack/migration-guide.1.2.12.md

```diff
@@ -0,0 +1,113 @@
+   - *-AzureRmOffer is replaced with *-AzsOffer
+   - *-AzureRmManagedOffer is replaced with *-AzsManagedOffer
+   - *-AzureRmPlan is replaced with *-AzsPlan
+   - *-AzureRmMangedSubscription is replaced with *-AzsUserSubscription
+   - *-AzureRMTenantSubscription is replaced with *-AzsSubscription
+   - *-AzureRmManagedLocation is replaced with *-AzsLocation
+
+**Deprecated Cmdlets**
+
+- ```Get-AzureStackToken``` is deprecated. If there is a need to get token, please use the Microsoft.IdentityModel.Clients.ActiveDirectory.dll directly.
+
+- The cmdlets ```*-AzsReourceProviderManifest``` and ```*-AzsUsageConnection``` are deprecated
+For any of these usages, template based resource creation is preferred.  For deletion of these resources, deleting the whole resource group is preferred. If needed, you can also use generic cmdlet *-AzureRmResource
+
+**Parameter Changes**<br>
+The previous release was printing  warnings about the deprecation of the following parameter aliases and they are deprecated now. 
+- Parameter alias ```PlanName``` has been deprecated in favor of ```Name``` in Plan cmdlets
+- Parameter alias ```OfferName``` has been deprecated in favor of ```Name``` in Offer cmdlets
+- Parameter alias ```ResourceGroup``` has been deprecated in favor of ```ResourceGroupName``` in all the cmdlets
+- Parameter ```Managed``` in Get-AzsPlan has been deprecated. It was not used before
+- Parameter ```Managed``` in Get-AzsOffer has been deprecated. Please use Get-AzsManagedOffer instead
+
+**Flattening of Properties** <br>
+The Plan/Offer/Subscription objects wont be having .Properties instead the child properties are flattened out to be at the Top level. If you are having any references to the .Properties, this breaking change could be fixed by accessing the prperty directly at the top level
+
+**Force Parameter** <br>
+
+Remove-* cmdlets will ask for confirmation before doing the remove action. Please use -Force to avoid the confirmation. The following are the affected cmldets
+- Remove-AzsOffer
+- Remove-AzsPlan
+- Remove-AzsUserSubscription
+- Remove-AzsSubscription
+
+<br>
+
+## Breaking Changes in Storage Admin Module
+
+**Globally Removed Parameters**<br>
+The parameters ```SkipCertificateValidation``` and ```DefaultProfile``` have been removed from all cmdlets.
+
+The parameters ```TimeGrain```, ```StartTimeInUtc```, ```EndTimInUtc```, ```MetricNames```, ```DetailedOutput```  have been removed for Get-Azs*Metric Cmdlets, please replace with Where-Object and Select-Object.
+
+The pareamters ```MetricNames```, ```DetailedOutput```  have been removed for Get-Azs*MetricDefinition Cmdlets, please replace with Where-Object and Select-Object
+
+**New Parameters**
+- The parameter ```ResourceGroupName``` is now an optional for all cmdlets.
```

Comment:
Except in subscriptions and AzureBridge!

Reply:
Okay!  Do we mention somewhere that all resource group names are optional except for those two?  Also, how do users find those resource groups?  I guess that is not for use to describe?

Reply:
don't think it needs to go in a separate doc, each cmdlet's reference doc should cover that